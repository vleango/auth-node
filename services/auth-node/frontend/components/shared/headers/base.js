import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

export const Header = ({ headerLink, onClickPressed, buttonText }) => (
  <div>
    <header>
      <Link to={headerLink}>
        <h3>App</h3>
      </Link>
      <button onClick={onClickPressed}>{buttonText}</button>
    </header>
  </div>
);

export default Header;
