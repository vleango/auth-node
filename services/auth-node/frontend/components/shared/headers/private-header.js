import React from 'react';
import { connect } from 'react-redux';

import Header from './base';
import { startLogout } from '../../../actions/auth/logout';

export const PrivateHeader = ({ startLogout }) => (
  <Header
    headerLink='/dashboard'
    buttonText='Logout'
    onClickPressed={startLogout} />
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: async () => await dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(PrivateHeader);
