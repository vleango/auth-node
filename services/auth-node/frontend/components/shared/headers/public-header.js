import React from 'react';

import Header from './base';

export class PublicHeader extends React.Component {

  loginBtnClicked = () => {
    this.props.history.push('/login');
  };

  render() {
    return (
      <Header
        headerLink='/'
        buttonText='Login'
        onClickPressed={this.loginBtnClicked} />
    );
  }

}

export default PublicHeader;
