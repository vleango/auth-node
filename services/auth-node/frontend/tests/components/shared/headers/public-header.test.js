import React from 'react';
import { shallow } from 'enzyme';

import { PublicHeader } from '../../../../components/shared/headers/public-header';

describe('Components', () => {
  describe('Shared', () => {
    describe('Headers', () => {
      describe('PublicHeader', () => {

        it('should render PublicHeader correctly', () => {
        	const wrapper = shallow(<PublicHeader />);
        	expect(wrapper).toMatchSnapshot();
        });

      });

    });
  });
});
