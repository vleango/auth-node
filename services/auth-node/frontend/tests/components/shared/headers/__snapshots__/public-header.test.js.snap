// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Shared Headers PublicHeader should render PublicHeader correctly 1`] = `
<Header
  buttonText="Login"
  headerLink="/"
  onClickPressed={[Function]}
/>
`;
